src/service/db.py

```diff
@@ -13,26 +13,44 @@ class SongServiceByDB(SongService):
     def __init__(
         self,
         googleCredentials: Mapping[str, str],
-        defaultSpreadsheetId: str = "",
+        spreadsheetIds: Mapping[str, str],
+        defaultGroup: str = "",
         cache: Optional[Cache] = None,
     ) -> None:
-        self.defaultSpreadsheetId = defaultSpreadsheetId
+        self.spreadsheetIds = spreadsheetIds
+        self.defaultSpreadsheetId = spreadsheetIds[defaultGroup]
         self.service = SongDB(googleCredentials, cache=cache)
         self.cache = cache
 
+        self._create_song_mappings()
+
+    def _create_song_mappings(self):
+        self.songMappings = {
+            self._to_song_key(song): {"group": group, "name": song}
+            for group, id in self.spreadsheetIds.items()
+            for song in self.service.list_song_names(id)
+        }
+
     @with_cache("SongServiceByDB::get_song")
-    def get_song(self, songName: str, spreadsheetId: str = "") -> Song:
-        if spreadsheetId == "":
-            spreadsheetId = self.defaultSpreadsheetId
+    def get_song(self, songName: str, spreadsheet: str = "") -> Song:
```

Comment:
Since we already have `group` as a concept in the `SongService` class, let's change the signature of all the `get_*` methods and `save_song` to accept a `group` instead of a `spreadsheetId`

Reply:
Wait, now that I think about it, `get_song` doesn't need a group argument... since the script can pull the group info from the song name, it's not relevant.

Unless there's an edge case where there's multiple songs with the same title by different groups?

Reply:
I thought about that too and yeah, I think it's fine to assume that there's no duplicate song title in different sheets

Reply:
fixed